azure/converters/futures_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package converters
+
+import (
+	"testing"
+
+	azureautorest "github.com/Azure/go-autorest/autorest/azure"
+	"github.com/onsi/gomega"
+
+	infrav1 "sigs.k8s.io/cluster-api-provider-azure/api/v1alpha4"
+)
+
+var (
+	validFuture = infrav1.Future{
+		Type:          infrav1.DeleteFuture,
+		ServiceName:   "test-service",
+		Name:          "test-group",
+		ResourceGroup: "test-group",
+		Data:          "eyJtZXRob2QiOiJERUxFVEUiLCJwb2xsaW5nTWV0aG9kIjoiTG9jYXRpb24iLCJscm9TdGF0ZSI6IkluUHJvZ3Jlc3MifQ==",
+	}
+
+	emptyDataFuture = infrav1.Future{
+		Type:          infrav1.DeleteFuture,
+		ServiceName:   "test-service",
+		Name:          "test-group",
+		ResourceGroup: "test-group",
+		Data:          "",
+	}
+
+	decodedDataFuture = infrav1.Future{
+		Type:          infrav1.DeleteFuture,
+		ServiceName:   "test-service",
+		Name:          "test-group",
+		ResourceGroup: "test-group",
+		Data:          "this is not b64 encoded",
+	}
+
+	invalidFuture = infrav1.Future{
+		Type:          infrav1.DeleteFuture,
+		ServiceName:   "test-service",
+		Name:          "test-group",
+		ResourceGroup: "test-group",
+		Data:          "ZmFrZSBiNjQgZnV0dXJlIGRhdGEK",
+	}
+)
+
+func Test_FutureToSDK(t *testing.T) {
```

Comment:
added a test, it's not very thorough because I can't fake an error when calling MarshalJSON() but at least it makes sure we set the right values, so it's better than nothing.

Reply:
yeah I don't think we can do more on this. Thanks for adding the happy path test.